public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -321,7 +321,9 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
     [runQueriesPreview, setValue, updateExpressionAndDatasource]
   );
 
-  const recordingRuleDefaultDatasource = rulesSourcesWithRuler[0];
+  // Using dataSourcesWithRuler[0] gives incorrect types - no undefined
+  // Using at(0) provides a safe type with undefined
+  const recordingRuleDefaultDatasource = rulesSourcesWithRuler.at(0);
```

Comment:
We could try to enable https://www.typescriptlang.org/tsconfig/#noUncheckedIndexedAccess

Reply:
💡